editor/src/messages/layout/utility_types/layout_widget.rs

```diff
@@ -34,85 +34,103 @@ pub enum Layout {
 	MenuLayout(MenuLayout),
 }
 
-impl Layout {
-	pub fn unwrap_widget_layout(self, action_input_mapping: &impl Fn(&MessageDiscriminant) -> Vec<KeysGroup>) -> WidgetLayout {
-		if let Layout::WidgetLayout(mut widget_layout) = self {
-			// Function used multiple times later in this code block to convert `ActionKeys::Action` to `ActionKeys::Keys` and append its shortcut to the tooltip
-			let apply_shortcut_to_tooltip = |tooltip_shortcut: &mut ActionKeys, tooltip: &mut String| {
-				let shortcut_text = tooltip_shortcut.to_keys(action_input_mapping);
-
-				if let ActionKeys::Keys(_keys) = tooltip_shortcut {
-					if !shortcut_text.is_empty() {
-						if !tooltip.is_empty() {
-							tooltip.push(' ');
-						}
-						tooltip.push('(');
-						tooltip.push_str(&shortcut_text);
-						tooltip.push(')');
+#[derive(PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub enum DiffUpdate {
+	#[serde(rename = "subLayout")]
+	SubLayout(SubLayout),
+	#[serde(rename = "layoutGroup")]
+	LayoutGroup(LayoutGroup),
+	#[serde(rename = "widget")]
+	Widget(WidgetHolder),
+}
+
+#[derive(PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub struct WidgetDiff {
+	pub path: Vec<usize>,
```

Comment:
Would `widget_path` be less ambiguous? We're trying to reduce the highly-overloaded nature of the word `path` throughout our codebase.